processor/resourcedetectionprocessor/internal/aws/ecs/ecs.go

```diff
@@ -0,0 +1,166 @@
+// TODO: Replace all attribute fields and enums with values defined in "conventions" once they exist
+func (d *Detector) Detect(context.Context) (pdata.Resource, error) {
+	res := pdata.NewResource()
+	res.InitEmpty()
+
+	tmde := getTmdeFromEnv()
+
+	// Fail fast if neither env var is present
+	if tmde == "" {
+		log.Println("No Task Metadata Endpoint environment variable detected, skipping ECS resource detection")
+		return res, nil
+	}
+
+	tmdeResp, err := d.provider.fetchTaskMetaData(tmde)
+
+	if err != nil || tmdeResp == nil {
+		return res, err
+	}
+
+	attr := res.Attributes()
+	attr.InsertString(conventions.AttributeCloudProvider, conventions.AttributeCloudProviderAWS)
+	attr.InsertString("cloud.infrastructure_service", "ECS")
+	attr.InsertString("aws.ecs.task.arn", tmdeResp.TaskARN)
+	attr.InsertString("aws.ecs.task.family", tmdeResp.Family)
+
+	// TMDE returns the the short name or ARN, so we need to parse out the short name from ARN if applicable
+	cluster := parseCluster(tmdeResp.Cluster)
+	attr.InsertString("aws.ecs.cluster", cluster)
+
+	region, account := parseRegionAndAccount(tmdeResp.TaskARN)
+	if account != "" {
+		attr.InsertString(conventions.AttributeCloudAccount, account)
+	}
+
+	if region != "" {
+		attr.InsertString(conventions.AttributeCloudRegion, region)
+	}
+
+	// The Availability Zone is not available in all Fargate runtimes
+	if tmdeResp.AvailabilityZone != "" {
+		attr.InsertString(conventions.AttributeCloudZone, tmdeResp.AvailabilityZone)
+	}
+
+	// The launch type and log data attributes are only available in TMDE v4
+	switch lt := strings.ToLower(tmdeResp.LaunchType); lt {
+	case "ec2":
+		attr.InsertString("aws.ecs.launchtype", "EC2")
+
+	case "fargate":
+		attr.InsertString("aws.ecs.launchtype", "Fargate")
+	}
+
+	selfMetaData, err := d.provider.fetchContainerMetaData(tmde)
+
+	if err != nil || selfMetaData == nil {
+		return res, err
+	}
+
+	logAttributes := [4]string{"aws.log.group.names", "aws.log.group.arns", "aws.log.stream.names", "aws.log.stream.arns"}
+
+	for i, attribVal := range getValidLogData(tmdeResp.Containers, selfMetaData, account) {
+		if attribVal.Len() > 0 {
+			ava := pdata.NewAttributeValueArray()
+			ava.SetArrayVal(attribVal)
+			attr.Insert(logAttributes[i], ava)
+		}
+	}
+
+	return res, nil
+}
+
+func getTmdeFromEnv() string {
+	var tmde string
+	if tmde = strings.TrimSpace(os.Getenv(tmde4EnvVar)); tmde == "" {
+		tmde = strings.TrimSpace(os.Getenv(tmde3EnvVar))
+	}
+
+	return tmde
+}
+
+func parseCluster(cluster string) string {
+	parts := strings.Split(cluster, "/")
+	if len(parts) > 1 {
+		return parts[1]
+	}
+
+	return cluster
+}
+
+// Parses the AWS Account ID and AWS Region from a task ARN
+// See: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-account-settings.html#ecs-resource-ids
+func parseRegionAndAccount(taskARN string) (region string, account string) {
+	parts := strings.Split(taskARN, ":")
```

Comment:
Not sure how important counting the colons is to follow the above advice here too

Reply:
Not really sure how to do this without making like 3 calls to `LastIndexByte`. I need to parse out the `account_id` and `region` from the Task ARN documented [here](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-account-settings.html#ecs-resource-arn-timeline), so counting colons matters I think.